test/unit/resources/json_test.rb

```diff
@@ -86,30 +86,61 @@
     end
   end
 
-  describe "#load_raw_from_file" do
-    let(:cmd_str)  { "curl localhost" }
-    let(:resource) { Inspec::Resources::JsonConfig.allocate }
-    let(:inspec)   { mock }
-    let(:command)  { mock }
+  describe ":command" do
+    # CASES:
+    #
+    # stdout:good,  stderr:empty
+    # stdout:bad,   stderr:empty
+    # stdout:empty, stderr:msg
+    # stdout:empty, stderr:empty
 
-    before do
-      resource.stubs(:inspec).returns(inspec)
-      inspec.expects(:command).with(cmd_str).returns(command)
+    # TODO: abstract and push up
+    def run_json_cmd(cmd)
+      backend = Inspec::Backend.create(Inspec::Config.new)
+      klass   = Inspec::Resource.registry["json"]
+
+      klass.new(backend, "json", command: cmd)
+    end
+
+    # TODO: push up
+    def assert_resource_skipped(res, msg)
+      assert_predicate res, :resource_skipped?
+      refute_predicate res, :resource_failed?
+
+      assert_includes res.resource_exception_message, msg
     end
 
-    it "raises an exception if command stdout is nil" do
-      command.expects(:stdout).returns(nil)
-      proc { resource.send(:load_raw_from_command, cmd_str) }.must_raise Inspec::Exceptions::ResourceSkipped
+    # TODO: push up
+    def assert_resource_failed(res, msg)
+      refute_predicate res, :resource_skipped?
+      assert_predicate res, :resource_failed?
+
+      assert_includes res.resource_exception_message, msg
     end
 
-    it "raises an exception if command stdout is empty" do
-      command.expects(:stdout).returns("")
-      proc { resource.send(:load_raw_from_command, cmd_str) }.must_raise Inspec::Exceptions::ResourceSkipped
+    it "good stdout, empty stderr" do
+      resource = run_json_cmd %(echo '{"result":true}')
+
+      assert_equal %({"result":true}\n), resource.raw_content
+      assert_equal({ "result" => true }, resource.params)
+    end
+
+    it "bad stdout, empty stderr" do
+      resource = run_json_cmd "echo 'not-valid-json'"
+
+      assert_resource_failed resource, "unexpected token at 'not-valid-json"
     end
 
-    it "returns the command output" do
-      command.expects(:stdout).returns("json goes here")
-      resource.send(:load_raw_from_command, cmd_str).must_equal "json goes here"
+    it "empty stdout, message in stderr" do
+      resource = run_json_cmd "echo bad args 1>&2 && false"
```

Comment:
Looks like the `&&` breaks the windows tests...

Reply:
uuuugh. THAT thing again. I'll figure it out.

Reply:
This is blocked by #4465, which I just filed.